### DISCUSSION TITLE
Typescript issue: Type is not assignable to type 'Encrypter'.

### DISCUSSION THREAD

**Discussion by NayamAmarshe** [digitalfortress-tech/localstorage-slim#19] [Q&A] [answered]
I have this snippet:
```ts
// LOCALSTORAGE-SLIM
ls.config.encrypter = (data: unknown, secret: string): string => {
  return CryptoJS.AES.encrypt(data, secret).toString();
};
ls.config.decrypter = (encryptedString, secret) => {
  const bytes = CryptoJS.AES.decrypt(encryptedString, secret);
  return bytes.toString(CryptoJS.enc.Utf8);
};
```
but typescript throws an error: `Type '(data: unknown, secret: string) => string' is not assignable to type 'Encrypter'.`

How can I fix this? Tried a few things but they don't seem to work.

**Comment by niketpathak** [comment, marked as answer]
Hello @NayamAmarshe , 

You need to use **type-assertion** since localstorage-slim does not have an idea about what you plan to use internally within the Encrypter/Decrypter functions (i.e. CryptoJS or some other encryption library)

If you try this, your compiler will not complain -> 

```js
ls.config.encrypter = (data, secret): string => {
  return CryptoJS.AES.encrypt((data as string).toString(), secret as string).toString();
};
ls.config.decrypter = (encryptedString, secret) => {
  const bytes = CryptoJS.AES.decrypt(encryptedString as string, secret as string);
  return bytes.toString(CryptoJS.enc.Utf8);
};
```

Here is a simple [article on type-assertion](https://bobbyhadz.com/blog/typescript-type-unknown-is-not-assignable-to-type) to clarify your doubts.

**Reply by NayamAmarshe** [reply]
Thanks a lot Niket!
I tried this and it fixed the issue:
```ts
ls.config.encrypter = (data, secret) => {
  return CryptoJS.AES.encrypt(
    JSON.stringify(data),
    secret as string
  ).toString();
};

ls.config.decrypter = (data, secret) => {
  try {
    return JSON.parse(
      CryptoJS.AES.decrypt(data as any, secret as any).toString(
        CryptoJS.enc.Utf8
      )
    );
  } catch (e) {
    console.log(e);
    return data;
  }
};
```

**Reply by niketpathak** [reply]
While your solution will work, its always better to use specific types rather using `any` since that kinda beats the purpose of using typescript.
